### PR TITLE
[nvmon] Flush on SIGTERM

### PR DESCRIPTION
**Pull request by Elarnon**
The benchmark scripts use nvmon to monitor the frequency during telamon
runs.  nvmon is killed at the end of the run, but currently does not
flush its output -- which means that when stdout is redirected to a
file, we lose the very data we wanted to capture.

This patch adds a SIGTERM handler to nvmon to flush `stdout` before
exiting.

### CODE REVIEW

File: tools/nvmon/nvmon.c [hunk 1]
```diff
@@ -5,12 +5,14 @@
...
 #include <assert.h>
-#include <nvml.h>
+#include <signal.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 
+#include <nvml.h>
+
```
Comment (andidr) [inline tools/nvmon/nvmon.c]: Is there a particular reason for moving this include down or was this change just committed by accident?

File: tools/nvmon/nvmon.c [hunk 2]
```diff
@@ -403,6 +410,16 @@ void print_samples(void) {
 
 int main(int argc, char **argv) {
   if (argc == 1) {
+    struct sigaction sa;
+    sa.sa_handler = on_sigterm;
+    sigemptyset(&sa.sa_mask);
+    sa.sa_flags = 0;
+
+    if (sigaction(SIGTERM, &sa, NULL) == -1) {
+      fprintf(stderr, "Failed to register SIGTERM handler");
```
Comment (andidr) [inline tools/nvmon/nvmon.c]: Since no string formatting takes place, `fputs` should be sufficient

File: tools/nvmon/nvmon.c (hunk 1 above)
Comment (Elarnon) [reply tools/nvmon/nvmon.c]: It was moved down to separate system includes from CUDA includes.

File: tools/nvmon/nvmon.c (hunk 2 above)
Comment (Elarnon) [reply tools/nvmon/nvmon.c]: I prefer using `fprintf` everywhere when performance is not critical for consistency and to make it easier to add string formatting if wanted.  I'm fine with changing it to `fputs` however.

Comment (andidr) [reply tools/nvmon/nvmon.c]: No strong feelings about this, but `fprintf` looks weird without formatting.

**Pull request merged**